File: padroesComportamentais/padrao-de-projeto-interpreter/php/src/Importadora.php
<?php

namespace Interpreter\Php;

use MathParser\Interpreting\Evaluator;
use MathParser\StdMathParser;

class Importadora
{
    public static string $formula = "(precoProduto + pesoProduto) * 2";

    public static function calcularImposto(float $preco, float $peso): float
    {
        $formula = str_replace(
            ["precoProduto", "pesoProduto"],
            [$preco, $peso],
            self::$formula
        );

        $mathParser = new StdMathParser();
        $evaluator = new Evaluator();
        $result = $mathParser->parse($formula);
        return $result->evaluate($evaluator);
    }
}

File: padroesComportamentais/padrao-de-projeto-interpreter/java/src/main/java/Importadora.java
public class Importadora {
    public static String formula = "preco + peso * 2";

    public static double calcularImposto(double preco, double peso) {
        String expressao;
        expressao = formula.replace("preco", Double.toString(preco));
        expressao = expressao.replace("peso", Double.toString(peso));
        InterpretadorExpressao interpretador = new InterpretadorExpressoesAritmeticas(expressao);
        return interpretador.interpretar();
    }
}
